projects/extension/src/background/AppMediator.ts

```diff
@@ -69,6 +75,30 @@ export class AppMediator {
   }
 
   processSmoldotMessage(message: JsonRpcResponse): boolean {
+    if (this.#state === 'disconnected') {
```

Comment:
add these in if () { } else if { } else.
This way you "save" some time for addition if checks (once 1 if succeeds rest checks wont follow)

Reply:
Could you elaborate please?  I don't understand.  This is a guard clause.. it returns early if the state is disconnected because it shouldn't be processing the message

Reply:
Misread the code here a bit - I was mentioning the way if is written but please disreagard this. Its my mistake